zad1/main.rb

```diff
@@ -1,13 +1,23 @@
 def sort_letters(string)
-
+p= string.chars.sort.join
+    return p
 end
 
 def count_vowels(string)
-
+    sum=0
+    (0...string.size).each do |i|
+        sum+=1 if string[i][/[aeiou]/]
+    end
+    return sum
 end
 
-def filter_odd(array)
-
+def filter_odd (array)
+    k=[]
+    j=-1;
+    (0...array.size).each do |i|
```

Comment:
Możemy się pozbyć zmiennej `k` przez wykorzystanie metody inject z prezentacji. Inject sam nam tworzy obiekt, który zwróci na końcu i wtedy `k` nam odpada. Można to oczywiście zrobić jeszcze ładniej, Array w Ruby posiada masę metod do właśnie takiego filtrowania np. [polecany select](https://ruby-doc.org/core-2.2.0/Array.html#method-i-select), ale Array jest też enumerablem więc ma posiada dużo metod do operacji na kolekcjach które można znaleźć [tutaj](https://ruby-doc.org/core-2.5.1/Enumerable.html)